Source/GUI/MainWindow.cpp

```diff
@@ -80,7 +81,10 @@ void MainWindow::makeMenus()
   m_actScanner = new QAction(tr("&Scanner"), this);
   m_actScanner->setShortcut(QKeySequence("F3"));
   m_actScanner->setCheckable(true);
-  m_actScanner->setChecked(m_splitter->sizes()[0] > 0);
+  QTimer::singleShot(0, [this]() {
```

Comment:
I'm assuming we dont have a way around not using QTimer?

Reply:
1. Another option would be to connect to the `aboutToShow` signal of the menu, and update its state every time.
2. Another option could be to force the visibility of the main window within the `MainWindow()` constructor [provided this can be done within the constructor], and then update the action's state before leaving the constructor.
3. Another option could be to check whether it's a fresh config and, if so, set the state to `true`, as we'd know no setting that would ultimately reset the splitter could be called.

I'm not sure 2. and 3. would be very elegant. 1. could work, provided the state of the action [which would be inconsistent until the menu is shown] is not relied upon [now or in the future].

(I'd probably stick with the single-shot timer to be honest.)

Reply:
I'm not well versed in Qt so I'll trust your judgement with the singleShot then